frigate/detectors/plugins/hailo8l.py

```diff
@@ -24,263 +28,357 @@
 
 from frigate.const import MODEL_CACHE_DIR
 from frigate.detectors.detection_api import DetectionApi
-from frigate.detectors.detector_config import BaseDetectorConfig
-
-# Set up logging
-logger = logging.getLogger(__name__)
-
-# Define the detector key for Hailo
+from frigate.detectors.detector_config import BaseDetectorConfig, ModelTypeEnum, InputTensorEnum, PixelFormatEnum, InputDTypeEnum
+from PIL import Image, ImageDraw, ImageFont
+
+# ----------------- Inline Utility Functions ----------------- #
+def preprocess_image(image: Image.Image, model_w: int, model_h: int) -> Image.Image:
+    """
+    Resize image with unchanged aspect ratio using padding.
+    """
+    img_w, img_h = image.size
+    scale = min(model_w / img_w, model_h / img_h)
+    new_img_w, new_img_h = int(img_w * scale), int(img_h * scale)
+    image = image.resize((new_img_w, new_img_h), Image.Resampling.BICUBIC)
+    padded_image = Image.new('RGB', (model_w, model_h), (114, 114, 114))
+    padded_image.paste(image, ((model_w - new_img_w) // 2, (model_h - new_img_h) // 2))
+    return padded_image
+
+def extract_detections(input_data: list, threshold: float = 0.5) -> dict:
+    """
+    (Legacy extraction function; not used by detect_raw below.)
+    Extract detections from raw inference output.
+    """
+    boxes, scores, classes = [], [], []
+    num_detections = 0
+    for i, detection in enumerate(input_data):
+        if len(detection) == 0:
+            continue
+        for det in detection:
+            bbox, score = det[:4], det[4]
+            if score >= threshold:
+                boxes.append(bbox)
+                scores.append(score)
+                classes.append(i)
+                num_detections += 1
+    return {
+        'detection_boxes': boxes,
+        'detection_classes': classes,
+        'detection_scores': scores,
+        'num_detections': num_detections
+    }
+# ----------------- End of Utility Functions ----------------- #
+
+# Global constants and default URLs
 DETECTOR_KEY = "hailo8l"
+ARCH = None
+H8_DEFAULT_MODEL = "yolov8s.hef"
+H8L_DEFAULT_MODEL = "yolov6n.hef"
+H8_DEFAULT_URL = "https://hailo-model-zoo.s3.eu-west-2.amazonaws.com/ModelZoo/Compiled/v2.14.0/hailo8/yolov8s.hef"
+H8L_DEFAULT_URL = "https://hailo-model-zoo.s3.eu-west-2.amazonaws.com/ModelZoo/Compiled/v2.14.0/hailo8l/yolov6n.hef"
+
+def detect_hailo_arch():
+    try:
+        result = subprocess.run(['hailortcli', 'fw-control', 'identify'], capture_output=True, text=True)
+        if result.returncode != 0:
+            print(f"Error running hailortcli: {result.stderr}")
```

Comment:
Any `print` statements should be a `logger.*` of some type so they are sent to the parent's log pipe.